core/src/main/java/io/confluent/connect/storage/util/DataUtils.java

```diff
@@ -16,32 +16,42 @@
 
 package io.confluent.connect.storage.util;
 
+import io.confluent.connect.storage.common.util.StringUtils;
 import org.apache.kafka.connect.data.Field;
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.errors.DataException;
 
 import java.util.Map;
 
 public class DataUtils {
 
   public static Object getField(Object structOrMap, String fieldName) {
+    validate(structOrMap, fieldName);
+
+    Object field;
     if (structOrMap instanceof Struct) {
-      return ((Struct) structOrMap).get(fieldName);
+      field = ((Struct) structOrMap).get(fieldName);
     } else if (structOrMap instanceof Map) {
-      Object field = ((Map<?, ?>) structOrMap).get(fieldName);
-      if (field == null) {
-        throw new DataException(String.format("Unable to find nested field '%s'", fieldName));
-      }
+      field = ((Map<?, ?>) structOrMap).get(fieldName);
+    } else {
+      throw new DataException(String.format(
+            "Argument not a Struct or Map. Cannot get field '%s' from %s.",
+            fieldName,
+            structOrMap
+      ));
     }
-    throw new DataException(String.format(
-          "Argument not a Struct or Map. Cannot get field '%s' from: %s",
-          fieldName,
-          structOrMap
-    ));
+    if (field == null) {
+      throw new DataException(
+            String.format("The field '%s' does not exist in %s.", fieldName, structOrMap));
+    }
```

Comment:
Added to prevent NPE in `TimestampExtractor`

```java
Object timestampValue = DataUtils.getNestedFieldValue(struct, fieldName);
Schema fieldSchema = DataUtils.getNestedField(record.valueSchema(), fieldName).schema();

if (Timestamp.LOGICAL_NAME.equals(fieldSchema.name())) {
  return ((Date) timestampValue).getTime(); // <------ here
}

switch (fieldSchema.type()) {
  case INT32:
  case INT64:
    return ((Number) timestampValue).longValue();  // <----- here
```